src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -495,6 +499,8 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
             if (__DEV__) {
               startPhaseTimer(finishedWork, 'componentDidMount');
             }
+            instance.props = finishedWork.memoizedProps;
+            instance.state = finishedWork.memoizedState;
```

Comment:
Is there any observable way these wouldn't already be the same?

Other than the user setting them themselves? (I suppose if they do, then this is a breaking change now.)